File: fase8-kvm/glkvm-cloud/internal/http/dto/personal.go
package dto

// ---- profile ----

type PersonalProfileResp struct {
	ID               int64  `json:"id"`
	Username         string `json:"username"`
	DisplayName      string `json:"displayName"`
	Email            string `json:"email"`
	Role             string `json:"role"`
	AuthProvider     string `json:"authProvider"`
	RegistrationTime int64  `json:"registrationTime"`
	LastLoginTime    *int64 `json:"lastLoginTime"`
	TotpEnabled      bool   `json:"totpEnabled"`
}

type UpdatePersonalProfileReq struct {
	DisplayName *string `json:"displayName"`
}

// ---- 2fa ----

type Setup2faResp struct {
	Secret     string `json:"secret"`
	OtpauthURL string `json:"otpauthUrl"`
}

type Enable2faReq struct {
	Secret string `json:"secret"`
	Code   string `json:"code"`
}

type Disable2faReq struct {
	Code string `json:"code"`
}

// ---- trusted devices ----

type TrustedDevice struct {
	ID         int64  `json:"id"`
	DeviceName string `json:"deviceName"`
	IP         string `json:"ip"`
	CreatedAt  int64  `json:"createdAt"`
	LastUsedAt int64  `json:"lastUsedAt"`
	ExpiresAt  int64  `json:"expiresAt"`
}

type ListTrustedDevicesResp struct {
	Items []TrustedDevice `json:"items"`
}

File: fase8-kvm/glkvm-cloud/ui/src/models/personal.ts
/**
 * Personal Center 类型声明
 */

import { AuthProviderEnum, UserRoleEnum } from './userManage'

export interface PersonalProfile {
    id: number
    username: string
    displayName: string
    email: string
    role: UserRoleEnum
    authProvider: AuthProviderEnum
    /** 注册时间，unix 秒，0 表示未知 */
    registrationTime: number
    /** 最近登录时间，unix 秒，null 表示从未登录 */
    lastLoginTime: number | null
    totpEnabled: boolean
}

export interface Setup2faResp {
    secret: string
    otpauthUrl: string
}

export interface TrustedDevice {
    id: number
    deviceName: string
    ip: string
    /** unix 秒 */
    createdAt: number
    /** unix 秒 */
    lastUsedAt: number
    /** unix 秒 */
    expiresAt: number
}

export interface TrustedDeviceList {
    items: TrustedDevice[]
}
